erpnext/hr/doctype/payroll_entry/payroll_entry.js

```diff
@@ -72,16 +73,99 @@ frappe.ui.form.on('Payroll Entry', {
 		})
 	},
 
+	update_salary_slips: function(frm) {
+		frm.call({
+			doc: frm.doc,
+			method: "update_salary_slips",
+		})
+	},
+
 	add_context_buttons: function(frm) {
 		if(frm.doc.salary_slips_submitted || (frm.doc.__onload && frm.doc.__onload.submitted_ss)) {
 			frm.events.add_bank_entry_button(frm);
 		} else if(frm.doc.salary_slips_created) {
+			frm.add_custom_button(__("Update Salary Slips"), function () {
+				frm.trigger("update_salary_slips");
+			});
+
 			frm.add_custom_button(__("Submit Salary Slip"), function() {
 				submit_salary_slip(frm);
 			}).addClass("btn-primary");
 		}
 	},
 
+	make_disbursement_entry: function(frm) {
+		frappe.call({
+			doc: frm.doc,
+			method: "get_disbursement_mode_details",
+			callback: function(r) {
+				if (r && r.message) {
+					let salary_modes = r.message[0]
+					let banks = r.message[1]
+					let d = new frappe.ui.Dialog({
+						title: 'Select Salary Mode',
+						fields: [
+							{
+								label: 'Payment Account',
+								fieldname: 'payment_account',
+								fieldtype: 'Link',
+								options: "Account",
+								reqd: 1,
+								get_query: function () {
+									return {
+										filters: {
+											company: frm.doc.company,
+											is_group: 0,
+											account_type: ['in', ['Cash', 'Bank']]
+										}
+									}
+								}
+							},
+							{
+								label: 'Salary Mode',
+								fieldname: 'salary_mode',
+								fieldtype: 'Select',
+								options: salary_modes
+							},
+							{
+								label: 'Bank',
+								fieldname: 'bank_name',
+								fieldtype: 'Select',
+								options: banks,
+								depends_on: "eval:doc.salary_mode == 'Bank'"
+							},
+						],
+						primary_action_label: 'Submit',
+						primary_action: function(r) {
+							return frappe.call({
+								doc: cur_frm.doc,
+								method: "make_payment_entry",
+								args: {
+									payment_account: r.payment_account,
+									salary_mode: r.salary_mode,
+									bank_name: r.bank_name
+								},
+								freeze: true,
+								freeze_message: __("Creating Payment Entries......"),
+								callback: function(r) {
+									if (r.message && r.message.length) {
+										if (r.message.length === 1) {
+											frappe.set_route('Form', 'Journal Entry', r.message[0]);
+										} else {
+											frappe.set_route('List', 'Journal Entry',
+												{"Journal Entry Account.reference_name": frm.doc.name});
+										}
+									}
+								}
+							});
+						}
+					});
+					d.show();
+				}
+			}
+		});
+	},
+
 	add_bank_entry_button: function(frm) {
 		frappe.call({
 			method: 'erpnext.hr.doctype.payroll_entry.payroll_entry.payroll_entry_has_bank_entries',
```

Comment:
You'll have to modify this API in order show the button even after Bank Entries exist to allow creating multiple bank entries